projects/ngx-monaco-tree/src/lib/monaco-tree-file/monaco-tree-file.component.ts

```diff
@@ -1,67 +1,152 @@
-  standalone: true,
-  imports: [NgIf, NgForOf, MonacoTreeContextMenuComponent, DragDropModule, NgStyle, CdkDrag, CdkDrag, CdkDropList],
+  imports: [
+    MonacoTreeContextMenuComponent,
+    DragDropModule,
+    NgClass,
+    NgStyle,
+    CdkDrag,
+    CdkDrag,
+    CdkDropList
+  ],
   templateUrl: './monaco-tree-file.component.html',
   styleUrls: ['./monaco-tree-file.component.scss']
 })
 export class MonacoTreeFileComponent implements OnChanges {
-	@Input() name = '';
-  @Input() path = '';
-  @Input() color?: string|null|undefined = '';
-	@Input() content: MonacoTreeElement[]|undefined|null = undefined;
-	@Input() depth = 0;
-	@Input() theme: 'vs-dark'|'vs-light' = 'vs-dark';
-	@Input() hide = false;
-  @Input() current: string|null = null;
-
-	@Output() clickFile = new EventEmitter<string>();
-	@Output() contextMenuClick = new EventEmitter<ContextMenuAction>();
-  @Output() dragDropFile = new EventEmitter<DragAndDropEvent>();
-
-  private children = viewChildren(MonacoTreeFileComponent);
-
-	open = false;
-	position: [number, number]|undefined = undefined;
-
+  private readonly eRef = inject(ElementRef);
+  private readonly children = viewChildren(MonacoTreeFileComponent);
+
+  readonly name = input.required<string>();
+  readonly path = input('');
+  readonly color = input<string>();
+  readonly content = input.required<MonacoTreeElement[] | undefined>();
+  readonly depth = input(0);
+  readonly theme = input<'vs-dark' | 'vs-light'>('vs-dark');
+
+  readonly current = model<string | null>(null);
+
+  readonly contextMenuClick = output<ContextMenuAction>();
+  readonly dragDropFile = output<DragAndDropEvent>();
+
+  readonly open = signal(false);
+  readonly position = signal<[number, number] | undefined>(undefined);
+
+  readonly fullPath = computed(() => `${this.path() ? this.path() + '/' : ''}${this.name()}`);
+  readonly isActive = computed(() => this.current() === this.fullPath());
+  readonly style = computed(() => `margin-left: ${10 * this.depth()}px`);
+  readonly folder = computed(() => Array.isArray(this.content()));
+  readonly icon = computed(() => {
+    if (this.folder()) {
+      const name = this.name();
+      if (Object.keys(folders).includes(name)) {
+        const icon = folders[name as keyof typeof folders];
+        if (this.open()) return icon + '-open'
+        else return icon;
+      }
+      else {
+        if (this.open()) return 'folder-open'
+        else return 'folder';
+      }
+    } else {
+      const name = this.name();
+      if (Object.keys(files).includes(name)) {
+        return files[name as keyof typeof files];
+      } else {
+        let splitted = name.split('.')
+        while (splitted.length > 0) {
+          splitted = splitted.slice(1)
+          const ext = splitted.join('.')
+          if (ext && Object.keys(extensions).includes(ext)) {
+            return extensions[ext as keyof typeof extensions];
+          }
+        }
+        return 'file'
+      }
+    }
+  });
```

Comment:
you may want to track all the input signals first in the function
if the signal is not tracked in the first call of the `computed`, it may not be properly tracked later
```suggestion
  readonly icon = computed(() => {
    const folder = this.folder();
    const open = this.open();
    const name = this.name();
    if (folder) {
      if (Object.keys(folders).includes(name)) {
        const icon = folders[name as keyof typeof folders];
        if (open) return icon + '-open'
        else return icon;
      }
      else {
        if (open) return 'folder-open'
        else return 'folder';
      }
    } else {
      if (Object.keys(files).includes(name)) {
        return files[name as keyof typeof files];
      } else {
        let splitted = name.split('.')
        while (splitted.length > 0) {
          splitted = splitted.slice(1)
          const ext = splitted.join('.')
          if (ext && Object.keys(extensions).includes(ext)) {
            return extensions[ext as keyof typeof extensions];
          }
        }
        return 'file'
      }
    }
  });
```

Reply:
I think we should keep it like it was because [computed signal dependencies are dynamic](https://angular.dev/guide/signals#computed-signal-dependencies-are-dynamic)